packages/ozone-typescript-client/src/taskClient/taskClientImpl.ts

```diff
@@ -54,50 +54,60 @@ export class TaskHandlerImpl<T = any> implements TaskHandler {
 
 	private _waitForSubTasks(asyncTasksGroupId: string): Promise<void> {
 		return (new Promise((resolve, reject) => {
-			this.subTaskinterval = window.setInterval(() => {
-				this._awaitTask(asyncTasksGroupId)
-					.then((data: GroupExecution) => {
-						if (data.stepsCount === data.stepsDone) {
-							this._clearPullInterval()
-							resolve()
-						} else if (data.hasErrors) {
-							reject('One of the processing sub tasks as an error')
- }
-					})
-					.catch((error) => {
-						this._clearPullInterval()
-						reject(error)
-					})
-			}, this.pollInterval)
+			const wait = () => {
+				this.subTimeout = window.setTimeout(() => {
+					this._awaitTask(asyncTasksGroupId)
+						.then((data: GroupExecution) => {
+							if (data.stepsCount !== data.stepsDone) {
+								wait()
```

Comment:
This cannot work: your original Promise will be forever pending.
Here is an example to illustrate the behavior I'm talking about. It should have resolved in 5 seconds
<img width="839" alt="image" src="https://user-images.githubusercontent.com/5877409/170259976-fdd0ee4e-0934-437f-8887-eb954136125a.png">

Here is an example that works
<img width="1092" alt="image" src="https://user-images.githubusercontent.com/5877409/170259837-b18de458-6283-4b98-aa13-6bd6549ca896.png">

Reply:
My bad, did not realize that the Promise was actually around the wait function. Nvm